File: src/lib/scala/Slate.Common/src/main/scala/slate/common/databases/DbCon.scala
/**
<slate_header>
  author: Kishore Reddy
  url: https://github.com/kishorereddy/scala-slate
  copyright: 2016 Kishore Reddy
  license: https://github.com/kishorereddy/scala-slate/blob/master/LICENSE.md
  desc: a scala micro-framework
  usage: Please refer to license on github for more info.
</slate_header>
  */

package slate.common.databases


trait DbCon {
  val driver:String
  val url:String
  val user:String
  val password:String
}


/**
  * Connection string for a database
  * @param driver   : jdbc driver
  * @param url      : url
  * @param user     : username
  * @param password : password
  */
final case class DbConString(driver:String, url:String, user:String, password:String)
  extends DbCon
{
}


/**
  * Empty connection string
  */
case object DbConEmpty extends DbCon {
  val driver:String   = ""
  val url:String      = ""
  val user:String     = ""
  val password:String = ""
}

File: src/lib/kotlin/slatekit-common/src/main/kotlin/slatekit/common/db/DbCon.kt
/**
 * <slate_header>
 * url: www.slatekit.com
 * git: www.github.com/code-helix/slatekit
 * org: www.codehelix.co
 * author: Kishore Reddy
 * copyright: 2016 CodeHelix Solutions Inc.
 * license: refer to website and/or github
 * about: A tool-kit, utility library and server-backend
 * mantra: Simplicity above all else
 * </slate_header>
 */

package slatekit.common.db

interface DbCon {
  val driver: String
  val url: String
  val user: String
  val password: String

  companion object {

    @JvmField val empty = DbConString("", "", "", "")
  }
}

/**
  * Connection string for a database
  * @param driver : jdbc driver
  * @param url : url
  * @param user : username
  * @param password : password
  */
data class DbConString(
    override val driver: String,
    override val url: String,
    override val user: String,
    override val password: String
) : DbCon

